pandas/compat/chainmap.py

```diff
@@ -15,3 +15,8 @@ def __delitem__(self, key):
                 del mapping[key]
                 return
         raise KeyError(key)
+
+    def new_child(self, *args, **kwargs) -> "DeepChainMap":
+        # ChainMap.new_child returns self.__class__(...) but mypy
+        #  doesn't know that, so we annotate it explicitly here.
+        return super().new_child(*args, **kwargs)  # type: ignore
```

Comment:
I _think_ this could be a typeshed issue not a mypy issue as new_child should return type of self not `typing.ChainMap[_KT, _VT]` see https://github.com/python/typeshed/blob/34d68ab0a2117a08fa221d3a10884f35cacf2cdc/stdlib/3/collections/__init__.pyi#L323-L338

Reply:
the `# type: ignore` could be avoided with

```python
    def new_child(self, *args, **kwargs) -> "DeepChainMap":
        result = super().new_child(*args, **kwargs)
        assert isinstance(result, DeepChainMap)
        return result
```